src/sidebar/config/build-settings.js

```diff
@@ -29,129 +31,156 @@ function getAncestorFrame(levels, window_ = window) {
         // The `groups` need to be fetched from a secondary RPC call and
         // there should be no timeout as it may be waiting for user input.
-        service.groups = rpcCall('requestGroups', [index], null).catch(() => {
-          throw new Error('Unable to fetch groups');
-        });
+        service.groups = postMessageJsonRpc
+          .call(
+            rpcSettings.targetFrame,
+            rpcSettings.origin,
+            'requestGroups',
+            [index],
+            0 // no timeout
+          )
+          .catch(() => {
+            throw new Error('Unable to fetch groups');
+          });
       }
     });
   }
-  return config;
+  return configFromHost;
+}
+
+/**
+ * Derive RPC settings from the provided `ConfigFromHost`, if any are present.
+ *
+ * @param {ConfigFromHost} configFromHost
+ * @param {Window} window_
+ * @return {import('../../types/config').RPCSettings|null}
+ */
+function buildRPCSettings(configFromHost, window_) {
+  const rpcConfig = configFromHost.requestConfigFromFrame;
+  if (!rpcConfig) {
+    return null;
+  } else if (
+    typeof rpcConfig.ancestorLevel !== 'number' ||
+    typeof rpcConfig.origin !== 'string'
+  ) {
+    throw new Error(
+      'Improper `requestConfigFromFrame` object. Both `ancestorLevel` and `origin` need to be specified'
+    );
+  }
+  return {
+    targetFrame: getAncestorFrame(rpcConfig.ancestorLevel, window_),
+    origin: rpcConfig.origin,
+  };
+}
+
+/**
+ * Retrieve ConfigFromHost to use for settings from the ancestor frame indicated
+ * in `rpcSettings`
+ *
+ * @param {ConfigFromHost} hostConfigFromURL
+ * @param {RPCSettings} rpcSettings
+ * @return {Promise<ConfigFromHost>}
+ */
+async function getHostConfigByRPC(hostConfigFromURL, rpcSettings) {
+  const hostConfigFromFrame = await postMessageJsonRpc.call(
+    rpcSettings.targetFrame,
+    rpcSettings.origin,
+    'requestConfig',
+    [],
+    3000
+  );
+
+  // In the case where the appropriate `ConfigFromHost` is sourced from another
+  // frame by RPC, the original `ConfigFromHost` (`hostConfigFromURL`) is
+  // discarded.
+  //
+  // The `group` property, however, is currently not available in the remote
+  // `ConfigFromHost` and needs to be restored. This property is used by the
+  // Notebook.
+  return {
+    ...hostConfigFromFrame,
+    ...(hostConfigFromURL.group ? { group: hostConfigFromURL.group } : {}),
+  };
 }
 
 /**
  * Build a `SidebarSettings` object by merging the provided `ConfigFromSidebar`
  * with `ConfigFromHost` from an appropriate source.
  *
  * `ConfigFromHost` may come from either:
- *  - The URL hash of the iframe, written by the annotator when creating the
- *    sidebar's iframe, OR
- *  - By sending an RPC request for host configuration to a known ancestor frame
+ *  - The URL hash of the sidebar's iframe src, written by the annotator when
+ *    creating the sidebar's iframe, OR
+ *  - By sending an RPC request for host configuration to a designated ancestor
+ *    frame
```

Comment:
I think we should make a note here that the LMS app is the main use case for this capability.